### PR TITLE
Env vars

### PR DESCRIPTION
**Pull request by bonndan**
This is an experiment to use labels to create relations between services.

### CODE REVIEW

File: src/main/java/de/bonndan/nivio/model/Items.java [hunk 1]
```diff
@@ -121,25 +121,29 @@ public static LandscapeItem pick(final String identifier, String group, final Co
...
 
         String query = "SELECT * FROM items WHERE " + term;
-        return cqnQuery(query, items);
+        return cqnQueryOnIndex(query, index(items));
     }
 
-    public static final Attribute<LandscapeItem, String> IDENTIFIER = attribute("identifier", LandscapeItem::getIdentifier);
-    public static final Attribute<LandscapeItem, String> NAME = attribute("name", LandscapeItem::getName);
+    private static final Attribute<LandscapeItem, String> IDENTIFIER = attribute("identifier", LandscapeItem::getIdentifier);
+    private static final Attribute<LandscapeItem, String> NAME = attribute("name", LandscapeItem::getName);
```
Comment (basseur) [inline src/main/java/de/bonndan/nivio/model/Items.java]: Move all `private static final` to the top.

File: src/main/java/de/bonndan/nivio/model/Items.java [hunk 2]
```diff
@@ -121,25 +121,29 @@ public static LandscapeItem pick(final String identifier, String group, final Co
...
-     * @return resultset
+     * @param items landscape items
+     * @return indexed collection
      */
-    private static List<? extends LandscapeItem> cqnQuery(String condition, Collection<? extends LandscapeItem> items) {
+    public static IndexedCollection<LandscapeItem> index(Collection<? extends LandscapeItem> items) {
+        IndexedCollection<LandscapeItem> index = new ConcurrentIndexedCollection<>();
+        index.addAll(items);
+        return index;
+
```
Comment (basseur) [inline src/main/java/de/bonndan/nivio/model/Items.java]: remove empty line

File: src/test/java/de/bonndan/nivio/input/dto/LabelProcessorTest.java [hunk 3]
```diff
@@ -68,4 +68,32 @@ public void relations() {
...
+        assertEquals(0, item.getLabels().size());
+
+        LabelProcessor.applyLabel(item, "_credentials_", "x");
+        assertEquals(0, item.getLabels().size());
+
+        LabelProcessor.applyLabel(item, "token", "x");
+        assertEquals(0, item.getLabels().size());
+
+        LabelProcessor.applyLabel(item, "a_token_", "x");
+        assertEquals(0, item.getLabels().size());
```
Comment (basseur) [inline src/test/java/de/bonndan/nivio/input/dto/LabelProcessorTest.java]: Replace `assertEquals(0, item.getLabels().size())` with `assertTrue(item.getLabels().isEmpty())` for better readability.

File: src/main/java/de/bonndan/nivio/input/MagicLabelRelations.java [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+ * Examines the labels of an item for parts that point to being an url and could point to targets in the landscape.
+ *
+ *
+ *
+ */
+public class MagicLabelRelations {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(MagicLabelRelations.class);
+
+    private static final List<String> urlParts = Arrays.asList("uri", "url", "host");
```
[the remark this answers is not in the document]
Comment (bonndan) [reply src/main/java/de/bonndan/nivio/input/MagicLabelRelations.java]: done

File: src/main/java/de/bonndan/nivio/model/Items.java (hunk 2 above)
Comment (bonndan) [reply src/main/java/de/bonndan/nivio/model/Items.java]: done

File: src/main/java/de/bonndan/nivio/model/Items.java (hunk 1 above)
Comment (bonndan) [reply src/main/java/de/bonndan/nivio/model/Items.java]: done

**Comment by bonndan** [comment]
There are some bigger changes now.

File: src/test/java/de/bonndan/nivio/input/dto/LabelProcessorTest.java (hunk 3 above)
Comment (bonndan) [reply src/test/java/de/bonndan/nivio/input/dto/LabelProcessorTest.java]: Here I prefer to have the exact expected number written down.

**Pull request merged**